### DISCUSSION TITLE
Come aggiungere un nuovo cliente solo se già non è presente in anagrafica

### DISCUSSION THREAD

**Discussion by angelarted** [fattureincloud/api#437] [Q&A] [answered]
Ciao,
sto cercando di aggiungere un nuovo cliente in Fatture in Cloud solo se non è già presente in anagrafica.
Il primo step dello zap è che faccio l'elenco dei clienti in Fatture in Cloud, ma già qui c'è qualcosa che non va perché mi mostra sempre solo un cliente, e poi ovviamente tutto non funziona perché fa il confronto solo con quel cliente e non con tutti i record.

Nello screenshot allegato potete vedere l'esito dello step "List Clients": già che mi scriva "A clients was sent to Fatture in Cloud" mi lascia perplessa, io vorrei una lista di clienti su cui fare dei filtri, nello specifico attraverso il codice fiscale che è un parametro univoco e quindi se il codice fiscale del compratore non corrisponde con nessun codice fiscale già presente nella lista, allora crea nuovo cliente.
Ma come si fa se lui mi tira fuori sempre solo il primo record?????

Please help!!!
grazie,

Simona

![Schermata 2024-04-15 alle 11 40 15](https://github.com/fattureincloud/api/assets/5982081/42aac6e8-83b2-476a-bd8a-919e802d50e9)

**Comment by emajo** [comment]
Ciao @angelarted, per estrarre il cliente voluto è necessario utilizzare il campo `Query for filtering the results`, ad esempio tax_code='XXXX' recupererà il cliente con tax_code XXXX, [qui](https://developers.fattureincloud.it/docs/basics/additional-capabilities/filter-results#wrench-building-a-query) abbiamo una guida su come costruire il filtro.

Grazie,
Emanuele.

**Comment by angelarted** [comment]
Ciao, grazie per la risposta.
Ho due domande aggiuntive:
- primo, se, e come, è poi possibile utilizzare questa query che mi hai mandato in Zapier... non saprei come applicarla su Zapier.
- secondo, non ho capito se questa query va fatta dopo avere chiamato il metodo List Clients (e non mi è chiaro perché quel metodo come risultato mi dà solo un cliente, a questa domanda non mi è stato risposto) oppure sostituisce il metodo perché basta che la query sia null e il filtro è fatto, si va a creare il cliente.

Ribadisco che io **non voglio estrarre un cliente**, io voglio **vedere se il cliente esiste** per capre se andare a crearne uno nuovo oppure no.

Grazie mille,
Simona

**Reply by valmoz** [reply, marked as answer]
Buongiorno @angelarted 
come indicato dal mio collega @emajo tutte le chiamate di tipo List (tra cui la List Clients) permettono di filtrare i risultati sulla base di una query.
Nel suo caso specifico, le basterà aggiungere il campo "Query for filtering the results" alla List Clients che sta effettuando ora, inserendo la query che le interessa.
![image](https://github.com/fattureincloud/api/assets/7254293/71b590d7-2bcf-4d82-b362-b696e1db6ecb)

Come da lei ipotizzato nel punto due, nel caso in cui non dovesse essere individuato alcun risultato potrà poi procedere all'inserimento del cliente in anagrafica.

A disposizione
Mauro

**Reply by angelarted** [reply]
Ho capito, grazie!! 
Vi farò sapere se sono riuscita a risolvere! Ma perché mi restituiva comunque solo il primo cliente della lista? Non dovrebbe farmeli vedere tutti? Per curiosità... 
grazie ancora

**Reply by valmoz** [reply]
Ciao @angelarted 
Non sono un esperto di Zapier, ma quanto pare sembra essere una limitazione dello strumento che sulla List vuole come risultato un solo elemento.

Mauro

**Reply by angelarted** [reply]
Grazie mille, credo proprio di aver risolto con i vostri consigli @valmoz @emajo

**Reply by valmoz** [reply]
Ottimo! 
Chiudo questa discussione, se necessario la riapra senza problemi!

Grazie
Mauro
